backend/pikpak/pikpak.go

```diff
@@ -0,0 +1,1749 @@
+		"Id":                user.Sub,
+		"Username":          user.Name,
+		"Email":             user.Email,
+		"PhoneNumber":       user.PhoneNumber,
+		"Password":          user.Password,
+		"Status":            user.Status,
+		"CreatedAt":         time.Time(user.CreatedAt).String(),
+		"PasswordUpdatedAt": time.Time(user.PasswordUpdatedAt).String(),
+	}, nil
+}
+
+// ------------------------------------------------------------
+
+// get an id of file or directory
+func (f *Fs) getID(ctx context.Context, path string) (id string, err error) {
+	path = strings.Trim(path, "/")
+	id, err = f.dirCache.FindDir(ctx, path, false)
+	if err != nil {
+		o, err := f.NewObject(ctx, path)
+		if err != nil {
+			return "", err
+		}
+		id = o.(fs.IDer).ID()
+	}
+	return id, nil
+}
+
+// add offline download task for url
+func (f *Fs) addURL(ctx context.Context, url, path string) (*api.Task, error) {
+	req := api.RequestNewTask{
+		Kind:       api.KindOfFile,
+		UploadType: "UPLOAD_TYPE_URL",
+		URL: &api.URL{
+			URL: url,
+		},
+		FolderType: "DOWNLOAD",
+	}
+	if parentID, err := f.dirCache.FindDir(ctx, path, false); err == nil {
+		req.ParentID = parentIDForRequest(parentID)
+		req.FolderType = ""
+	}
+	return f.requestNewTask(ctx, &req)
+}
+
+type decompressDirResult struct {
+	Decompressed  int
+	SourceDeleted int
+	Errors        int
+}
+
+func (r decompressDirResult) Error() string {
+	return fmt.Sprintf("%d error(s) while decompressing - see log", r.Errors)
+}
+
+// decompress file/files in a directory of an ID
+func (f *Fs) decompressDir(ctx context.Context, filename, id, password string, srcDelete bool) (r decompressDirResult, err error) {
+	_, err = f.listAll(ctx, id, api.KindOfFile, "false", func(item *api.File) bool {
+		if item.MimeType == "application/zip" || item.MimeType == "application/x-7z-compressed" || item.MimeType == "application/x-rar-compressed" {
+			if filename == "" || filename == item.Name {
+				res, err := f.requestDecompress(ctx, item, password)
+				if err != nil {
+					err = fmt.Errorf("unexpected error while requesting decompress of %q: %w", item.Name, err)
+					r.Errors++
+					fs.Errorf(f, "%v", err)
+				} else if res.Status != "OK" {
+					r.Errors++
+					fs.Errorf(f, "%q: %d files: %s", item.Name, res.FilesNum, res.Status)
+				} else {
+					r.Decompressed++
+					fs.Infof(f, "%q: %d files: %s", item.Name, res.FilesNum, res.Status)
+					if srcDelete {
+						derr := f.deleteObjects(ctx, []string{item.ID}, f.opt.UseTrash)
+						if derr != nil {
+							derr = fmt.Errorf("failed to delete %q: %w", item.Name, derr)
+							r.Errors++
+							fs.Errorf(f, "%v", derr)
+						} else {
+							r.SourceDeleted++
+						}
+					}
+				}
+			}
+		}
+		return false
+	})
+	if err != nil {
+		err = fmt.Errorf("couldn't list files to decompress: %w", err)
+		r.Errors++
+		fs.Errorf(f, "%v", err)
+	}
+	if r.Errors != 0 {
+		return r, r
+	}
+	return r, nil
+}
+
+var commandHelp = []fs.CommandHelp{{
+	Name:  "getid",
```

Comment:
I wouldn't have thought you would need this - you can do this with `rclone lsf` or `rclone lsjson` which can show the IDs of files and directories.